dms/models/ir_attachment.py

```diff
@@ -54,21 +51,41 @@ def _dms_operations(self):
                 directories = self._get_dms_directories(
                     attachment.res_model, attachment.res_id
                 )
-            # Auto-create_files
+            # Auto-create_files (if not exists)
             for directory in directories:
-                self.env["dms.file"].sudo().create(
-                    {
-                        "name": attachment.name,
-                        "directory_id": directory.id,
-                        "attachment_id": attachment.id,
-                        "res_model": attachment.res_model,
-                        "res_id": attachment.res_id,
-                    }
+                dms_file_model = self.env["dms.file"].sudo()
+                dms_file = dms_file_model.search(
```

Comment:
Why is this called several times? The code this way is uglier... Maybe you can add a constraint and try to create always, although the ideal thing is to not call this several times.

Reply:
Anyway, shouldn't it be enough to search by directory and attachment?

Reply:
Now it's possible to enter in these function when `create` and `write` attachment. It's very important to check if `dms.file` exists or will be create 2 times (create and write). Checking if directory exists (according to record, sale for example) is not enough to check if `dms.file` is need to create.

Reply:
Searching for directory id + attachment id isn't enough? An attachment can be linked only to one record, so no duplicates.